app/controllers/concerns/hyrax/controller.rb

```diff
@@ -41,12 +41,12 @@ def search_results(*args)
 
   ##
   # @note for Blacklight 6/7 compatibility
-  def search_service(**search_params)
+  def search_service(*search_params)
     return super if defined?(super) && search_params.empty?
 
     search_service_class.new(config: blacklight_config,
                              scope: self,
-                             user_params: search_params,
+                             user_params: search_params.first,
```

Comment:
Are you happy with this part at least? I think it will fix a few different specs, and looks cleaner than what I came up with:
```
def search_results(*args)
    return super if defined?(super) # use the upstream if present (e.g. in BL 6)

    search_service(**args.reduce({}, :merge)).search_results
  end
```

Reply:
i think this might be better: https://github.com/samvera/hyrax/pull/6039

but i'm unsure if it will make any tests pass, or just make them fail on the PermissionQuery code.

Reply:
#6042 fixes the other half of this! it turns out there *are* unit tests for that behavior, and they were failing. i just hadn't spotted them.